client/layout/masterbar/woo-core-profiler.tsx

```diff
@@ -16,36 +16,27 @@ import getCurrentRoute from 'calypso/state/selectors/get-current-route';
 
 // Masterbar for WooCommerce Core Profiler Jetpack step
 const WooCoreProfilerMasterbar = ( { translate }: { translate: ( text: string ) => string } ) => {
-	const { redirectTo, shouldShowProgressBar, shouldShowNoThanks, currentRoute } = useSelector(
-		( state ) => {
-			const currentRoute = getCurrentRoute( state );
-			let redirectTo = null;
-			let shouldShowProgressBar = true;
-			let shouldShowNoThanks = true;
-			switch ( currentRoute ) {
-				case '/jetpack/connect/authorize':
-					redirectTo = getCurrentQueryArguments( state )?.redirect_after_auth;
-				case '/log-in/jetpack':
-					redirectTo = getQueryArg( getRedirectToOriginal( state ) || '', 'redirect_after_auth' );
-				default:
-			}
+	const currentQueryArguments = useSelector( getCurrentQueryArguments );
+	const currentRoute = useSelector( getCurrentRoute );
+	const redirectToOriginal = useSelector( getRedirectToOriginal );
 
-			if (
-				currentRoute === '/log-in/jetpack/lostpassword' ||
-				getCurrentQueryArguments( state )?.lostpassword_flow
-			) {
-				shouldShowProgressBar = false;
-				shouldShowNoThanks = false;
-			}
+	let redirectTo = null;
+	let shouldShowProgressBar = true;
+	let shouldShowNoThanks = true;
+	switch ( currentRoute ) {
+		case '/jetpack/connect/authorize':
+			redirectTo = currentQueryArguments?.redirect_after_auth;
+		case '/log-in/jetpack':
```

Comment:
Isn't there a missing `break;` statement? The `/jetpack/connect/authorize` case will fall through to the `/log-in/jetpack` case otherwise. The old code does that too, but it looks _very_ suspicious!

Reply:
Seems wrong. Adding some `break` statements in 315e9757da5b6409d86085f4881ae56bcb207692. Good catch!